externals/kyuubi-hive-sql-engine/src/main/scala/org/apache/kyuubi/engine/hive/events/HiveEngineEvent.scala

```diff
@@ -30,8 +30,14 @@ case class HiveEngineEvent(
     diagnostic: String,
     settings: Map[String, String]) extends KyuubiEvent {
 
-  override def partitions: Seq[(String, String)] =
-    ("day", Utils.getDateFromTimestamp(startTime)) :: Nil
+  override def partitions: Seq[(String, String)] = {
+    // before engine is started, the start time is 0L, the partition use current day.
+    if (startTime == 0) {
```

Comment:
if a engine is not started  how can we go into this code path , is it a bug ?

Reply:
After the engine is initialized, an event is posted.
https://github.com/apache/incubator-kyuubi/blob/14f675d28841cb7e049af187fcd1a339e24ffc6d/externals/kyuubi-hive-sql-engine/src/main/scala/org/apache/kyuubi/engine/hive/HiveSQLEngine.scala#L96

Reply:
so shall we also fix that code path ?

Reply:
do we cancel post event before engine start?

Reply:
I think we can. I looked at Spark engine, it also does not post before start

Reply:
Spark engine post an initialized event, its start time is start time of sparkContext

https://github.com/apache/incubator-kyuubi/blob/93f13ef662f497a0e02281cf09cbf8cd0d8577fd/externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala#L203

Reply:
ok, we can make a global variable at HiveEngineEvent, something like:
```scala
object HiveEngineEvent {
  val startTime = System.currentTimeMillis()
```

Reply:
The startTime has been existed in AbstractService, it is set in start method.

Reply:
In hive engine we can simplify to use this startTime, so any event can hold the same start time.

Reply:
add a engineStartTime